ciao-launcher/main.go

```diff
@@ -237,15 +236,35 @@ func (client *agentClient) ErrorNotify(err ssntp.Error, frame *ssntp.Frame) {
 	glog.Infof("ERROR %d", err)
 }
 
-func (client *agentClient) installLauncherDeps() {
-	role := client.conn.Role()
-	osprepare.Bootstrap()
-	if role.IsNetAgent() {
-		osprepare.InstallDeps(launcherNetNodeDeps)
-	}
-	if role.IsAgent() {
-		osprepare.InstallDeps(launcherComputeNodeDeps)
+func (client *agentClient) installLauncherDeps(doneCh chan struct{}) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	ch := make(chan error)
+	go func() {
+
+		osprepare.Bootstrap(ctx)
+
+		role := client.conn.Role()
+		if role.IsNetAgent() {
+			osprepare.InstallDeps(ctx, launcherNetNodeDeps)
+		}
+		if role.IsAgent() {
+			osprepare.InstallDeps(ctx, launcherComputeNodeDeps)
+		}
+
+		ch <- nil
+	}()
+
+	select {
+	case <-doneCh:
+		glog.Info("Received terminating signal.  Installation of launcher dependencies cancelled.")
```

Comment:
The way the code is written, there is no guarantee that the inner go routine we have spawned has exitted by the time we exit installLauncherDeps.  We can ensure this by doing


```
case <-doneCh:
  cancelFunc()
  <-ch
```
If we do this we don't need the call to cancelFunc at the bottom.